store/tikv/region_cache.go

```diff
@@ -226,7 +226,7 @@ func (c *RegionCache) ListRegionIDsInKeyRange(bo *Backoffer, startKey, endKey []
 			return nil, errors.Trace(err)
 		}
 		regionIDs = append(regionIDs, curRegion.Region.id)
-		if curRegion.Contains(endKey) {
+		if (startKey!=nil && curRegion.Contains(endKey)) || len(curRegion.EndKey) == 0 {
```

Comment:
Is it possible that  startKey is not nil and curRegion.start_key is nil?

Reply:
I suppose it should be:
if (**endKey** != nil && curRegion.Contains(endKey) || len(curRegion.EndKey) == 0 {

Reply:
I think this is the bug of `Contains` method.
Fix it there is better.
If a region has nil end key, it contains every key.

Reply:
I suppose it should be:
```
if (endKey != nil && curRegion.Contains(endKey) ||
   (endKey==nil && len(curRegion.EndKey)==0 ) {
...
}
```

Reply:
I do not agree with @coocood  since function `Contains` can not distinguish the key is `startKey` or `endKey`